powerline/listers/i3wm.py

```diff
@@ -0,0 +1,79 @@
+# vim:fileencoding=utf-8:noet
+from __future__ import (unicode_literals, division, absolute_import, print_function)
+
+import re
+
+from powerline.theme import requires_segment_info
+from powerline.lib.shell import run_cmd
+from powerline.lib.dict import updated
+
+
+conn = None
+XRANDR_OUTPUT_RE = re.compile('^([0-9A-Za-z-]+) connected')
```

Comment:
If you meant my comment regarding separate function: I do not see why can’t you just write

```Python
def get_connected_xrandr_outputs(pl):
	return (match.groupdict() for match in re.finditer(
		'^(?P<name>[0-9A-Za-z-]+) connected (?P<width>\d+)x(?P<height>\d+)\+(?P<x>\d+)\+(?P<y>\d+)',
		run_cmd(pl, ['xrandr', '-q']),
		re.MULTILINE
	))
```

and in lister simply write `output=output['name']`.